### ISSUE TITLE
Gradient Clipping

### ISSUE THREAD

**Issue by mfaisal59** [google/automl#910]
Hi,

I have one confusion, why are you clipping each variable norm followed by global norm. Only `clip_by_global_norm` should fulfill the purpose? why do we need to apply `clip_by_norm` first?
From `det_model_fn.py`
```
        # First clip each variable's norm, then clip global norm.
        clip_norm = abs(params['clip_gradients_norm'])
        clipped_grads = [
            tf.clip_by_norm(g, clip_norm) if g is not None else None
            for g in grads
        ]
        clipped_grads, _ = tf.clip_by_global_norm(clipped_grads, clip_norm)
```

**Comment by mingxingtan** [comment]
Hi @mfaisal59

While dealing with a special dataset, I found some variables can have large gradient norm, and therefore added this per-variable clipping. However, it turns out such per-variable clipping was not much helpful.

You can safely ignore these for most of dataset (like coco, pascal, etc).  I probably will delete it soon.

**Comment by faisal-kt** [comment]
Thanks @mingxingtan
